flask_resources/serializers/txt.py

```diff
@@ -0,0 +1,28 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2023 CERN.
+#
+# Flask-Resources is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""TXT serializer."""
+
+from .base import SerializerMixin
+
+
+class TextSerializer(SerializerMixin):
+    """TXT serializer implementation."""
+
+    def __init__(self, string_encoder):
+        """Initialize the TextSerializer."""
+        self._string_encoder = string_encoder
+
+    def serialize_object(self, obj, **kwargs):
+        """Dump the object into a string."""
+        return self._string_encoder(obj)
+
+    def serialize_object_list(self, obj_list, **kwargs):
+        """Dump the object list into a string."""
+        # this is a coupled to the `BaseListSchema` or any other list schema that
+        # returns its search results with this schema
+        return "\n".join([self.serialize_object(obj, **kwargs) for obj in obj_list])
```

Comment:
question: this list would come from a search so we would need the `[hits][hits]` accesing the dict no? in which case this becomes identical to the XMLSerializer. 

What is the main difference between pure text and xml serializer? if all the diff is made by the string encoder, then maybe we need to generalize (e.g. rename the XML one) and use it. WDYT?

Reply:
😂 I like the names. However, I think it should be more generic, maybe even `Base...` since today is xml and txt, tomorrow could be something else, and we should not need to re-rename it.

Before we dive into this, could you verify if this needs also the`[hits][hits]` accessing?

Reply:
Yes it is [hits][hits]
And your proposition does make sense

would `BaseSerializer` make sense as a name?

Reply:
Yes, I wonder if we should just rename `string_encoder` by `encoder` and deprecate the [SerializerMixin](https://github.com/inveniosoftware/flask-resources/blob/7a5ec0208f5e177188149a82b9916f3cb752d4aa/flask_resources/serializers/base.py#L13).

`SerializerMixin` is not being used as mixin anywhere, if so as interface. In that case it should be refactored to be an abstract class (i.e. inherit `ABC`) and have abstract methods instead of raising `NotImplemented`. This could be named `BaseSerializer`, then we could rename the new `TXTXML_Serializer` to `SimpleSerializer`, since it is not "base" but just very _simple_ in implementation/behaviour.

Reply:
Sounds good to me :)
`I can do it`
Would you want me to implement this or are you taking care of it? 
And if needed I'm free this afternoon starting 14h (classes got cancelled)

Reply:
I'll ping Karolina during the standup to get this going. By `I can do it` I meant the releases, the PR is yours ;)

Reply:
Cool I'll get to work on it then thx!

Reply:
we will discuss this soon, however there are naming changes in the classes so following the semver, it is not a patch release. Could you remove the release commit please? I think it is a bit premature to keep it here before we decide about the approach

Reply:
It keeps adding itself for some reason :/
I'll try and remove it again